mlrun/runtimes/serving.py

```diff
@@ -293,16 +296,35 @@ def set_tracking(
         batch: int = None,
         sample: int = None,
         stream_args: dict = None,
+        tracking_policy: dict = None,
     ):
-        """set tracking stream parameters:
+        """set tracking parameters:
+
+        :param stream_path:     Path/url of the tracking stream e.g. v3io:///users/mike/mystream
+                                you can use the "dummy://" path for test/simulation.
+        :param batch:           Micro batch size (send micro batches of N records at a time).
+        :param sample:          Sample size (send only one of N records).
+        :param stream_args:     Stream initialization parameters, e.g. shards, retention_in_hours, ..
+        :param tracking_policy: Dictionary that will be converted into a tracking policy object. By using
+                                TrackingPolicy, the user can apply his model monitoring requirements, such as setting
+                                the scheduling policy of the model monitoring batch job or changing the image of the
+                                model monitoring stream.
+
+                                example::
+                                # initialize a new serving function
+                                serving_fn = mlrun.import_function("hub://v2_model_server", new_name="serving")
+                                # apply model monitoring and set monitoring batch job to run every 3 hours
+                                tracking_policy = {'default_batch_intervals':"0 */3 * * *"}
+                                serving_fn.set_tracking(tracking_policy=tracking_policy)
 
-        :param stream_path:  path/url of the tracking stream e.g. v3io:///users/mike/mystream
-                             you can use the "dummy://" path for test/simulation
-        :param batch:        micro batch size (send micro batches of N records at a time)
-        :param sample:       sample size (send only one of N records)
-        :param stream_args:  stream initialization parameters, e.g. shards, retention_in_hours, ..
         """
+
+        # Applying model monitoring configurations
         self.spec.track_models = True
+        self.spec.tracking_policy = model_monitoring.TrackingPolicy()
+        if tracking_policy:
+            for key in tracking_policy:
+                setattr(self.spec.tracking_policy, key, tracking_policy[key])
```

Comment:
this is kinda weird, shouldn't the `TrackingPolicy` has a `@classmethod` that construct a tracking policy from a dictionary? will be more readable and logic.